### PR TITLE
Allowed more flexible usage of docs generator by supporting variable injections via `snippetOptions.definitions`

### LINKED ISSUE #1525: [Dev] Enable environment setup replacement in docs building process
It will allow reusing the `snippetadapter.js` file in other umberto projects without much hacking.

The snippet adapter should be able to replace variables (aka preprocess variables) in the compilation step so we'll be able to build docs with different environment setups.

### PR DESCRIPTION
**Pull request by ma2ciek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Other: Allowed dynamic environment configuration setup for docs by supporting variable injections via `snippetOptions.definitions`. Closes #1525.

---

### Additional information

*For example – encountered issues, assumptions you had to make, other affected tickets, etc.*

### CODE REVIEW

File: scripts/docs/snippetadapter.js [hunk 1]
```diff
@@ -27,7 +27,8 @@ module.exports = function snippetAdapter( data ) {
 		entry: data.snippetSource.js,
 		outputPath,
 		language: snippetConfig.language,
-		production: data.options.production
+		production: data.options.production,
+		templateCommentLoaderConfig: data.options.templateCommentLoaderConfig || {}
```
Comment (Reinmar) [inline scripts/docs/snippetadapter.js]: Why not `smarCodeInjectus`?

Comment (Reinmar) [reply scripts/docs/snippetadapter.js]: smart{CodeInject}uś

Comment (pjasiun) [reply scripts/docs/snippetadapter.js]: `injectedVariables`

File: scripts/docs/template-comment-loader.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+/* eslint-env node */
+
+/*
+ * Matches `// @if param // someCode` and evaluates the `option.param`.
+ * If the param is evaluated to true, then uncomments the code.
+ *
+ * Note: Use this only for conditional imports. Otherwise use the webpack define plugin.
+ */
+module.exports = function templateCommentLoader( source, map ) {
```
Comment (pjasiun) [inline scripts/docs/template-comment-loader.js]: `codeInjector`?

File: scripts/docs/snippetadapter.js (hunk 1 above)
Comment (pjasiun) [reply scripts/docs/snippetadapter.js]: But I do not think that many JS developers will find C++ names familiar.

File: scripts/docs/snippetadapter.js [hunk 3]
```diff
@@ -117,7 +118,8 @@ function getWebpackConfig( config ) {
 			new webpack.BannerPlugin( {
 				banner: bundler.getLicenseBanner(),
 				raw: true
-			} )
+			} ),
+			new webpack.DefinePlugin( config.definitions )
```
Comment (pjasiun) [inline scripts/docs/snippetadapter.js]: Maybe we should do `JSON.stringify` for each value automatically?

Comment (ma2ciek) [reply scripts/docs/snippetadapter.js]: Hm. We can do it.

Comment (ma2ciek) [reply scripts/docs/snippetadapter.js]: Actually, It's weird for me that this plugin doesn't do it out of the box.

**Pull request merged**